builtin/credential/jwt/path_oidc.go

```diff
@@ -240,23 +243,28 @@ func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request,
 		return logical.ErrorResponse(errLoginFailed + " Expired or missing OAuth state."), nil
 	}
 
+	deleteRequest := true
+	defer func() {
```

Comment:
Nice!

Should we have timeout/expiry on these and clean them as part of the [periodic function](https://github.com/openbao/openbao/blob/b7526c7f719b0dec0cc7c9cc9b12fbc211fe57fa/sdk/framework/backend.go#L63-L71)? Or is there some other check we need to do to ensure the OIDC request is still valid / isn't resuming an old request that should've expired?

Reply:
That's a good observation.  I had to dig into it.  It turns out that the requests are stored as a cache implemented by https://github.com/patrickmn/go-cache which automatically cleans items out after they expire.  If a poll is successfully completed, they get deleted earlier in the poll function.  So I think this is OK.

Reply:
Ah, smart. Sorry, I had thought this was backed by storage but it isn't.

That is curious, do you run a HA cluster and have you noticed any OIDC auth failures when the cluster changes leadership at all?

Reply:
Another very good observation.  I do run HA clusters but I don't think I explicitly tried switching leadership in the middle of an OIDC authentication.  I tried it now and indeed it gets a failure of `Vault login failed. Expired or missing OAuth state.`

But since oidc authentications are manual operations and HA cluster leadership switches are pretty rare, I think it's an acceptable behavior.  People will just retry if it fails.  I did not introduce the use of the go-cache, that has been the implementation at least before I started looking at it..

Reply:
Right, perfect.

Thanks @DrDaveD!